app/models/cms/email_template.rb

```diff
@@ -109,7 +109,7 @@ def published
     super or file&.read
   end
 
-  def save(*)
+  def save(...)
```

Comment:
is not original working here? The difference is that in the original approach arguments are ignored. And I think this is what the intention was.

Reply:
no parenthesis at all is also an alternative, if I'm not mistaken. I can't find any solid documentation about asterisk vs. triple dot, allI've read is that they are the same?

Reply:
You cant pass the `*` to a method call unlike `...`. But in this case seem equivalent.

Reply:
I couldn't find a good (or any actually) explanation in Ruby docs either... But `(*)`, `(...)` and no parenthesis are actually not equivalent...

Consider the following example:

```
class Parent
  def call(*args, **kwargs)
    puts "args:"
    puts args
    puts "kwargs:"
    puts kwargs
  end
end

class AsteriskChild < Parent 
  def call(*)
    super
  end
end

class DotsChild < Parent 
  def call(...)
    super
  end
end

class NoParenthesisChild < Parent 
  def call
    super
  end
end
```

```
irb(main):027:0> AsteriskChild.new.call(kwarg: 'hello')
args:
{:kwarg=>"hello"}
kwargs:
{}
=> nil
irb(main):028:0> DotsChild.new.call(kwarg: 'hello')
args:
kwargs:
{:kwarg=>"hello"}
=> nil
irb(main):029:0> 
irb(main):030:0> NoParenthesisChild.new.call(kwarg: 'hello')
(irb):23:in `call': wrong number of arguments (given 1, expected 0) (ArgumentError)
```

If there are only positional arguments, then the results for `*` and `...` are the same:
```
AsteriskChild.new.call('a','b','c')
DotsChild.new.call('a','b','c')

args:
a                                                                                      
b                                                                                      
c                                                                                      
kwargs:                                                                                
{}
```


There are actually multiple `save` methods in ActiveRecord (some for validated models, some for others), I think the main one is [this one]( https://github.com/rails/rails/blob/v6.1.7.8/activerecord/lib/active_record/persistence.rb#L473), which as I mentioned before expects keyword arguments, so I think `(*)` is not correct (prior to Rails 6 the `save` method definition was [different](https://github.com/rails/rails/blob/v5.2.8.1/activerecord/lib/active_record/persistence.rb#L274))

In any case, I think `(...)` is safer, as you don't need to care about the method being overriden or anything, as it is "universal".

Reply:
> In any case, I think (...) is safer, as you don't need to care about the method being overriden or anything, as it is "universal".

To me, the expected output would have ben:
```
args:
{:kwarg=>"hello"}
kwargs:
{}
```

But I didn't know there's separation between args and keyword args in ruby 3. For anybody else like me, take a look at [Separation of positional and keyword arguments in Ruby 3.0](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/)

Therefore I agree with using dots `(...)`.